Android/app/src/main/java/app/intra/sys/AnalyticsEvent.java

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2019 Jigsaw Operations LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package app.intra.sys;
+
+import android.content.Context;
+import android.os.Bundle;
+import com.google.firebase.analytics.FirebaseAnalytics;
+import java.lang.reflect.Method;
+
+/**
+ * Wrapper for Firebase Analytics event reporting.  In addition to improving the reporting API,
+ * this class also attaches the Mobile Country Code (MCC) to each event, which indicates the client
+ * country, to compensate for country classification errors in Firebase.
+ */
+public class AnalyticsEvent {
+  // Analytics events
+  public enum Events {
+    BOOTSTRAP,
+    BOOTSTRAP_FAILED,
+    BYTES,
+    EARLY_RESET,
+    STARTVPN,
+    TLS_PROBE,
+    TRY_ALL_ACCEPTED,
+    TRY_ALL_CANCELLED,
+    TRY_ALL_DIALOG,
+    TRY_ALL_FAILED,
+    TRY_ALL_REQUESTED,
+    UDP,
+  }
+
+  // Parameters sent in Analytics events
+  public enum Params {
+    BYTES,
+    CHUNKS,
+    DOWNLOAD,
+    DURATION,
+    LATENCY,
+    MCC,
```

Comment:
We need to be clearer about what MCC you are referring to: the connected network or the home network? Let's make that clear in the names.

Reply:
I changed the code to only report when they agree, to minimize confusion.